src/dispatching/update_listeners/webhooks.rs

```diff
@@ -99,12 +141,49 @@ where
     req.payload_mut().certificate = certificate.take();
     req.payload_mut().max_connections = max_connections;
     req.payload_mut().drop_pending_updates = Some(drop_pending_updates);
+    req.payload_mut().secret_token = Some(secret);
 
     req.send().await?;
 
     Ok(())
 }
 
+/// Generates a random string consisting of 32 characters (`a-z`, `A-Z`, `0-9`,
+/// `_` and `-`).
+fn gen_secret_token() -> String {
+    use rand::{distributions::Uniform, Rng};
+    const CHARSET: &[u8] = b"abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_-";
+    const SECRET_LENGTH: usize = 32;
+
+    let random = rand::thread_rng()
+        .sample_iter(Uniform::new(0, CHARSET.len()))
+        .map(|idx| CHARSET[idx] as char)
+        .take(SECRET_LENGTH);
+
+    let mut secret = String::with_capacity(SECRET_LENGTH);
+    secret.extend(random);
+
+    secret
+}
+
+fn check_secret(bytes: &[u8]) -> Result<&[u8], &'static str> {
+    let len = bytes.len();
+
+    // Check that length is in bounds
+    if !(1..=256).contains(&len) {
+        return Err("secret token length must be in range 1..=256");
+    }
+
+    // Check that all characters of the secret are supported by telegram
```

Comment:
I don't quite get the purpose of these comments, given that the information they communicate is already present in the code.

Reply:
Fair enough. Should I remove them?

Reply:
Yes, please.

Reply:
Done in b849067378853f654fc04bbdc00cb8ed1001a0ae